chainerx_cc/chainerx/routines/manipulation_test.cc

```diff
@@ -1,952 +0,0 @@
-#include "chainerx/routines/manipulation.h"
-
-#include <algorithm>
-#include <cstdint>
-#include <memory>
-#include <string>
-#include <vector>
-
-#include <gtest/gtest.h>
-#include <nonstd/optional.hpp>
-
-#include "chainerx/array.h"
-#include "chainerx/array_index.h"
-#include "chainerx/axes.h"
-#include "chainerx/backward.h"
-#include "chainerx/check_backward.h"
-#include "chainerx/device_id.h"
-#include "chainerx/error.h"
-#include "chainerx/shape.h"
-#include "chainerx/strides.h"
-#include "chainerx/testing/array.h"
-#include "chainerx/testing/array_check.h"
-#include "chainerx/testing/device_session.h"
-#include "chainerx/testing/routines.h"
-#include "chainerx/testing/threading.h"
-
-namespace chainerx {
-namespace {
-
-class ManipulationTest : public ::testing::TestWithParam<std::string> {
-protected:
-    void SetUp() override {
-        const std::string& backend_name = GetParam();
-        device_session_.emplace(DeviceId{backend_name, 0});
-    }
-
-    void TearDown() override { device_session_.reset(); }
-
-private:
-    nonstd::optional<testing::DeviceSession> device_session_;
-};
-
-TEST_THREAD_SAFE_P(ManipulationTest, AsScalar) {
-    using T = float;
-    T value = 2.0f;
-    Array a = testing::BuildArray({1, 1, 1}).WithData<T>({value}).WithPadding(1);
-
-    Run([&a, &value]() {
-        Scalar s = AsScalar(a);
-        EXPECT_EQ(s.kind(), GetKind(TypeToDtype<T>));
-        EXPECT_EQ(static_cast<T>(s), value);
-    });
-}
-
-TEST_P(ManipulationTest, AsScalarInvalidZeroElement) {
-    Array a = testing::BuildArray({0}).WithData<float>({});
-    EXPECT_THROW(AsScalar(a), DimensionError);
-}
-
-TEST_P(ManipulationTest, AsScalarInvalidMoreThanOneElements) {
-    Array a = testing::BuildArray({2}).WithData<float>({1.0f, 2.0f});
-    EXPECT_THROW(AsScalar(a), DimensionError);
-}
-
-TEST_THREAD_SAFE_P(ManipulationTest, RollAxis) {
```

Comment:
`RollAxis` does not have Python bindings is nor tested. I think it's better to keep these tests.

Reply:
Added Python binding.

Reply:
Thanks, it LGTM!